pkg/operator/bootstrapteardown/waitforceo.go

```diff
@@ -59,11 +60,28 @@ func waitForEtcdBootstrap(ctx context.Context, operatorRestClient rest.Interface
 	return nil
 }
 
-func done(etcd *operatorv1.Etcd) (bool, error) {
-	if operatorv1helpers.IsOperatorConditionTrue(etcd.Status.Conditions, "EtcdRunningInCluster") {
-		klog.Info("Cluster etcd operator bootstrapped successfully")
-		return true, nil
+func done(etcd *operatorv1.Etcd) bool {
+	scalingStrategy := operatorv1helpers.FindOperatorCondition(etcd.Status.Conditions, "BootstrapScalingStrategy")
+	if scalingStrategy != nil {
+		// for DelayedHAScalingStrategy we need to test whether the bootstrap member was already removed
+		// this is to ensure we properly remove the bootstrap member before it comes back as a proper control plane node
+		if scalingStrategy.Message == string(ceohelpers.DelayedHAScalingStrategy) {
+			if operatorv1helpers.IsOperatorConditionTrue(etcd.Status.Conditions, "EtcdBoostrapMemberRemoved") {
+				return true
+			}
+
+			klog.Infof("waiting on condition %s in etcd CR %s/%s to be True.", "EtcdBoostrapMemberRemoved", etcd.Namespace, etcd.Name)
+			return false
+		}
+
+		if operatorv1helpers.IsOperatorConditionTrue(etcd.Status.Conditions, "EtcdRunningInCluster") {
```

Comment:
I've yet to measure the impact on normal HA installations with this. As this presumably will delay the bootstrap process a bit. EtcdRunningInCluster is set way earlier than the bootstrap strategy status.